Text-field dynamic attributes

alien1976 opened the pull request:
Added documentation, tests and updated demo about the dynamic attributes

- [x] Self-reviewed and fixed any obvious mistakes, TODOs, removed debugging code, logs, etc
- [x] Updated package.json in related components, the library or the cli
- [x] Created tests
- [x] Tested in a browser
- [x] Tested in Gameface
- [x] Created or updated the documentation
- [ ] Transferred or updated the Hugo documentation and example
- [x] Added/updated demo in index.html in root ot the repository

Tony-CL reviewed (approved):
* Can we add these properties to the `.case` selector in the `<style>` tag in the **demo.html**?

```css
        .case {
            margin: 5px;
            width: 100%;
            padding-right: 10px;
        }
```
![image](https://github.com/CoherentLabs/GameUIComponents/assets/21058220/d067113d-30d5-4197-a99b-1a507e17e8cd)

* Can't set the `name` attribute with `document.querySelector('gameface-text-field').name = 'test'`. Add it to the documentation if added.
![image](https://github.com/CoherentLabs/GameUIComponents/assets/21058220/99301d72-6839-418d-a27e-1c6c091d435e)

venetak commented on components/text-field/script.js:
```diff
@@ -53,147 +76,315 @@ class TextField extends CustomElementValidator {
+     * @param {string|boolean} newValue
+     */
+    attributeChangedCallback(name, oldValue, newValue) {
+        if (!this.isRendered) return;
```
Kind of. I had the same issue with the dropdown.
The attrChangedCbb is called when the element it attached to the DOM but BEFORE the connectedCallback of the custom element as it is not upgraded yet. So the flow is:
1. element connected to DOM
2. attrChanged 
3. custom element upgraded
4. connected callback

venetak commented on components/text-field/script.js:
```diff
@@ -53,147 +76,315 @@ class TextField extends CustomElementValidator {
+     */
+    // eslint-disable-next-line max-lines-per-function
+    updateAttributeState(name, value) {
+        switch (name) {
```
Nice.

alien1976 commented:
> * Can we add these properties to the `.case` selector in the `<style>` tag in the **demo.html**?
> 
> ```css
>         .case {
>             margin: 5px;
>             width: 100%;
>             padding-right: 10px;
>         }
> ```
> 
> ![image](https://user-images.githubusercontent.com/21058220/272250429-d067113d-30d5-4197-a99b-1a507e17e8cd.png)
> 
> * Can't set the `name` attribute with `document.querySelector('gameface-text-field').name = 'test'`. Add it to the documentation if added.
>   ![image](https://user-images.githubusercontent.com/21058220/272267524-99301d72-6839-418d-a27e-1c6c091d435e.png)

I fixed the name property that is now working.

Why do we need this CSS? It will make the fields to have 100% width which is weird IMO.

alien1976 commented:
> > * Can we add these properties to the `.case` selector in the `<style>` tag in the **demo.html**?
> > 
> > ```css
> >         .case {
> >             margin: 5px;
> >             width: 100%;
> >             padding-right: 10px;
> >         }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ![image](https://user-images.githubusercontent.com/21058220/272250429-d067113d-30d5-4197-a99b-1a507e17e8cd.png)
> > 
> > * Can't set the `name` attribute with `document.querySelector('gameface-text-field').name = 'test'`. Add it to the documentation if added.
> >   ![image](https://user-images.githubusercontent.com/21058220/272267524-99301d72-6839-418d-a27e-1c6c091d435e.png)
> 
> I fixed the name property which is not working.
> 
> Why do we need this CSS? It will make the fields to have 100% width which is weird IMO.

@Tony-CL I fixed the UI as well.

Also I found a bug about search text field that I also fixed. The bug was before my changes but we didn't hit it.
